Add fixes and vanilla testing for Adagrad non-vectorized and vectorized optimizers to handle complex numbers

choward232 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #66672
* __->__ #66671

Made changes in the step function of the vectorized and non-vectorized adagrad optimizers to handle complex numbers as two real numbers as per 65711 on github

Differential Revision: [D31673503](https://our.internmc.facebook.com/intern/diff/D31673503/)

cc @vincentqb @jbschlosser @albanD @ezyang @anjali411 @dylanbespalko @mruberry @Lezcano @nikitaved

Commit:
[Pytorch][Bootcamp] Add fixes and vanilla testing for Adagrad non-vectorized and vectorized optimizers to handle complex numbers

Made changes in the step function of the vectorized and non-vectorized adagrad optimizers to handle complex numbers as two real numbers as per 65711 on github

Differential Revision: [D31673503](https://our.internmc.facebook.com/intern/diff/D31673503/)

[ghstack-poisoned]

albanD reviewed (commented):
Looks quite good. The CI needs fixing though (in particular lint)

Commit:
Update on "[Pytorch][Bootcamp] Add fixes and vanilla testing for Adagrad non-vectorized and vectorized optimizers to handle complex numbers"


Made changes in the step function of the vectorized and non-vectorized adagrad optimizers to handle complex numbers as two real numbers as per 65711 on github

Differential Revision: [D31673503](https://our.internmc.facebook.com/intern/diff/D31673503/)

[ghstack-poisoned]

albanD reviewed (approved):
LGTM

Commit:
Update on "[Pytorch][Bootcamp] Add fixes and vanilla testing for Adagrad non-vectorized and vectorized optimizers to handle complex numbers"


Made changes in the step function of the vectorized and non-vectorized adagrad optimizers to handle complex numbers as two real numbers as per 65711 on github

Differential Revision: [D31673503](https://our.internmc.facebook.com/intern/diff/D31673503/)

[ghstack-poisoned]

Commit:
Update on "Add fixes and vanilla testing for Adagrad non-vectorized and vectorized optimizers to handle complex numbers"


Made changes in the step function of the vectorized and non-vectorized adagrad optimizers to handle complex numbers as two real numbers as per 65711 on github

Differential Revision: [D31673503](https://our.internmc.facebook.com/intern/diff/D31673503/)

cc vincentqb jbschlosser albanD ezyang anjali411 dylanbespalko mruberry @Lezcano nikitaved

[ghstack-poisoned]

Commit:
Update on "Add fixes and vanilla testing for Adagrad non-vectorized and vectorized optimizers to handle complex numbers"


Made changes in the step function of the vectorized and non-vectorized adagrad optimizers to handle complex numbers as two real numbers as per 65711 on github

Differential Revision: [D31673503](https://our.internmc.facebook.com/intern/diff/D31673503/)

cc vincentqb jbschlosser albanD ezyang anjali411 dylanbespalko mruberry @Lezcano nikitaved

[ghstack-poisoned]
